omod/src/main/java/org/bahmni/module/hip/web/controller/WellnessRecordController.java

```diff
@@ -0,0 +1,63 @@
+package org.bahmni.module.hip.web.controller;
+
+import org.bahmni.module.hip.web.client.ClientError;
+import org.bahmni.module.hip.web.model.BundledWellnessResponse;
+import org.bahmni.module.hip.web.model.WellnessRecordBundle;
+import org.bahmni.module.hip.web.service.WellnessRecordService;
+import org.bahmni.module.hip.web.utils.DateUtils;
+import org.codehaus.jackson.map.ObjectMapper;
+import org.openmrs.module.webservices.rest.web.RestConstants;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.util.StringUtils;
+import org.springframework.web.bind.annotation.*;
+
+import java.io.IOException;
+import java.util.Date;
+import java.util.List;
+
+@RequestMapping(value = "/rest/" + RestConstants.VERSION_1 + "/hip/wellnessRecord")
+@RestController
+public class WellnessRecordController {
+	private WellnessRecordService wellnessRecordService;
+	private final ObjectMapper mapper = new ObjectMapper();
+	@Autowired
+	public WellnessRecordController (WellnessRecordService wellnessRecordService) {
+		this.wellnessRecordService = wellnessRecordService;
+	}
+
+	@RequestMapping(method = RequestMethod.GET, value = "/visit", produces = MediaType.APPLICATION_JSON_VALUE)
+	public @ResponseBody
+	ResponseEntity<?> getWellnessForVisit(@RequestParam String patientId,
+											   @RequestParam String visitUuid,
+											   @RequestParam(required = false) String fromDate,
```

Comment:
if you make the parameter type as Date, then you dont have to parse the date. Its ok, you are validating the date subsequently ..